dftimewolf/lib/module.py

```diff
@@ -72,19 +77,21 @@ def SetupLogging(self, threaded: bool = False) -> None:
 
       self.logger.addHandler(console_handler)
 
-  def LogStats(self, stats: Dict[str, Any]) -> None:
-    """Saves useful runtime statistics to the state for later processing.
+  def LogTelemetry(self, data: Dict[str, str]) -> None:
+    """Logs useful telemetry using the telemetry attribute in the state object.
 
     Args:
-      stats: Stats to store. Contents are arbitrary, but keys must be strings.
+      data: Key-value telemetry to store.
 
     Raises:
-      ValueError: If the keys in the stats dict are not strings.
+      ValueError: If the keys in the telemetry dict are not strings.
     """
-    if not all (isinstance(key, str) for key in stats.keys()):
-      raise ValueError("Stats keys must be strings.")
-    stastentry = state_lib.StatsEntry(type(self).__name__, self.name, stats)
-    self.state.StoreStats(stastentry)
+    if not all (isinstance(key, str) for key in data.keys()):
```

Comment:
My rationale for enforcing it was that I didn't want the database to be corrupted by bad datatypes. Spanner does indeed enforce this (so we'd get an exception from the spanner libs if we ever tried to store, say, an int), but one could imagine a scenario where the backend is more liberal, and if pytype happens to not to catch it, then it would pass onto the database. 

Given that we "soft enforce" it using datatypes, I think it's a good idea to make sure that we raise an exception earlier rather than later.

Reply:
On second though, I agree, let's remove the enforcment. Types should be enough.